src/controllers/userController.js

```diff
@@ -56,6 +56,43 @@ export const createUser = async (req, res) => {
   }
 };
 
+export const getUser = async (req, res) => {
+  const userId = req.params.id;
+
+  try {
+    const user = await User.findById(userId);
+    if (!user) {
+      logger.warn('User not found', {
+        method: req.method,
+        url: req.originalUrl,
+        user: userId,
+        userId: req.userId,
+        ip: req.ip,
+      });
+      return res.status(404).json({ message: 'User not found' });
+    }
```

Comment:
I'm going to remove this code because there is already a middleware that checks if the user exists.